### PR TITLE
.txt panel export all fields

### LINKED ISSUE #4925: Updating a panel gene, sets Reduced penetrance and Mosaicism fields to False
For instance I've added a comment to a gene in the panel:

image

When I export it looks like this, these 2 booleans and no comment:

image

### LINKED ISSUE #4928: Export all fields from a panel, not just the first 4 columns
Exporting a gene panel to .txt or .pdf from the panel page doesn't download the whole thing and you get just the first columns of the panel

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Closes #4928

### NOTE THAT THIS PR DOESN't FIX #4925 and #4930

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test - locally **:
1. In a local panel, edit one gene, for instance adding a comment, or inheritance models etc. Save.
2. Go to the updated panel, panel page. Export to txt
3. On main branch, notice that a limited number of fields are exported
4. On this branch all fieds are exported for the edited gene.
5. Make sure this PR doesn't disrupt the CLI panel export -> `scout --demo export panel -b 37  panel1 `and `scout --demo export panel -b 37 --bed panel1
`
**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by DN, CR

### CODE REVIEW

**Comment by northwestwitch** [comment]
I've just realised that there is a set of problems with the export of panels (see my recent list of issue) - I'll try to address one by one in separate PRs. Opening this for review!

**Comment by dnil** [comment]
Nice! I had a quick look; I think all good stuff, but I guess we better have a look at the downstream use of the panel export as well, and that will have to wait for tomorrow.

**Comment by northwestwitch** [comment]
> Nice! I had a quick look; I think all good stuff, but I guess we better have a look at the downstream use of the panel export as well,

The downstream use of the panel is basically fixing the import of comments and manual inheritance -> #2516, I was doing it in a separate PR

**Comment by dnil** [comment]
> > Nice! I had a quick look; I think all good stuff, but I guess we better have a look at the downstream use of the panel export as well,
> 
> The downstream use of the panel is basically fixing the import of comments and manual inheritance -> #2516, I was doing it in a separate PR

I think the crucial thing is this call here, right? https://github.com/Clinical-Genomics/cg/blob/560003d2766116e41c314a355847fe70cb38f319/cg/apps/scout/scoutapi.py#L52 
And then the question becomes if parsing of that is affected in any way!

**Comment by northwestwitch** [comment]
> I think the crucial thing is this call here, right? https://github.com/Clinical-Genomics/cg/blob/560003d2766116e41c314a355847fe70cb38f319/cg/apps/scout/scoutapi.py#L52
> And then the question becomes if parsing of that is affected in any way!

Yes, the call of that function should happen if yoy run `scout --demo export panel -b 37  panel1` and `scout --demo export panel -b 37 --bed panel1 `

**Comment by dnil** [comment]
Yes, the pure text panel output is also used from cg for getting genes in another place, not much further along:
https://github.com/Clinical-Genomics/cg/blob/560003d2766116e41c314a355847fe70cb38f319/cg/apps/scout/scoutapi.py#L89
I think the parsing looks ok, but we need to test this in integration.

**Comment by dnil** [comment]
> Yes, the pure text panel output is also used from cg for getting genes in another place, not much further along: https://github.com/Clinical-Genomics/cg/blob/560003d2766116e41c314a355847fe70cb38f319/cg/apps/scout/scoutapi.py#L89 I think the parsing looks ok, but we need to test this in integration.

Meh, it just uses the first two columns anyway. Let's make that a small integration test. 😆

File: scout/constants/panels.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+EXPORT_PANEL_FIELDS = [
```
Comment (dnil) [inline scout/constants/panels.py:1]: 👍

File: scout/export/panel.py [hunk 2]
```diff
@@ -111,28 +113,30 @@ def export_panels(adapter, panels, versions=None, build="37"):
...
 
-def export_gene_panels(adapter, panels, version=None):
+def set_export_fields(panels: list[str]) -> list[tuple]:
+    """Set the field that should be exported for one panel or a list of panels."""
+
+    if len(panels) == 1:
+        return EXPORT_PANEL_FIELDS
+
+    # if more than one panel is exported (from the CLI) - export only HGNC and symbol
+    return EXPORT_PANEL_FIELDS[:2]
```
Comment (dnil) [inline scout/export/panel.py:123]: ..and that ensure the automation sees the same thing as before for almost every order. We still should test what happens if CG was to ask for a run with a single panel on a case. But as long as the first two columns stay the same, everything should be just fine.

**Review by dnil — APPROVED** [review]
👍 Nice - will be useful to be able to save all values to e.g. other customers! As far as I can tell this both solves the issue, and is backwards compatible and safe to automation in the parts of the files it uses.

We should probably still test at couple of run starts from CG with one and many panels in the order to be sure we are not mistaken? 
My current feeling is that could be either here or for the next release, but maybe good to have it done: I'm pretty sure I can pull off a test with multiple panels, but will need to look up more to get a single panel one going.

**Comment by northwestwitch** [comment]
> We should probably still test at couple of run starts from CG with one and many panels in the order to be sure we are not mistaken?
> My current feeling is that could be either here or for the next release, but maybe good to have it done: I'm pretty sure I can pull off a test with multiple panels, but will need to look up more to get a single panel one going.

I've tested locally with more than one panel and it looked fine. Let's merge and test from cg before the next release

**Pull request merged**